src/Week3/Golf.hs

```diff
@@ -0,0 +1,133 @@
+module Week3.Golf where
+
+{-
+ Exercice 1: Hopscotch
+
+ Your first task is to write a function
+    skips :: [a] -> [[a]]
+
+ The output of skips is a list of lists. The first list in the output should
+ be the same as the input list. The second list in the output should
+ contain every second element from the input list. . . and the nth list in
+ the output should contain every nth element from the input list.
+
+ For example:
+  skips "ABCD" == ["ABCD", "BD", "C", "D"]
+  skips "hello!" == ["hello!", "el!", "l!", "l", "o", "!"]
+  skips [1] == [[1]]
+  skips [True,False] == [[True,False], [False]]
+  skips [] == []
+
+ Note that the output should be the same length as the input
+-}
+
+-- We make sure the element always exists by bounding the indices from n-1 to length list - 1
+-- We increment by n each time
+takeNth :: [a] -> Int -> [a]
+-- This doesn't feel much concise. Perhaps there is way to do this
+-- using filter and cycle.
+takeNth xs n = [xs !! i | i <- [n - 1, n - 1 + n..length xs - 1]]
+
+
+skips :: [a] -> [[a]]
+skips [] = []
+skips xs = map (takeNth xs) [1..(length xs)]
```

Comment:
Food for thought: what's the difference here between using `map` and `fmap`?

Reply:
It can be implemented using `fmap` as well. I have included that as a function `skips'`.
A very straightforward distinction between a `map` and an `fmap` would be that `map` function can be applied to nothing more than list of values. But `fmap` function can be applied much more data types. E.g. maybes, tuples, lists, etc.

And since here we are only applying a function to a list of `Int`, using `fmap` would also be accurate.